### PR TITLE
Rpl fixes

### PR DESCRIPTION
**Pull request by rveerama1**
RPL node link metric access was wrong, which resulting in invalid rank (same as base rank) for the node. Which is causing global repair from Border router when border router reboots. Root cause is rpl code trying to access link_metric variable from RPL parent nbr table, but it exists in IPv6 nbr table data.

### CODE REVIEW

File: subsys/net/ip/rpl-mrhof.c [hunk 1]
```diff
@@ -171,6 +164,7 @@ int net_rpl_of_neighbor_link_cb(struct net_if *iface,
 static u16_t calculate_path_metric(struct net_rpl_parent *parent)
 {
 	struct net_nbr *nbr;
+	struct net_ipv6_nbr_data *data;
```
Comment (tbursztyka) [inline subsys/net/ip/rpl-mrhof.c]: revert these 2 lines

File: subsys/net/ip/rpl.c [hunk 2]
```diff
@@ -281,6 +281,23 @@ struct net_nbr *net_rpl_get_nbr(struct net_rpl_parent *data)
 	return NULL;
 }
 
+struct net_ipv6_nbr_data *net_rpl_get_ipv6_nbr_data(
+						struct net_rpl_parent *parent)
+{
+	struct net_nbr *nbr;
+	struct net_nbr *ipv6_nbr;
```
Comment (tbursztyka) [inline subsys/net/ip/rpl.c]: declare ipv6_nbr after if (nbr)

File: subsys/net/ip/rpl.h [hunk 3]
```diff
@@ -306,6 +306,16 @@ struct net_rpl_dag {
 struct net_nbr *net_rpl_get_nbr(struct net_rpl_parent *data);
 
 /**
+ * @brief Get related neighbor data from parent pointer.
+ *
+ * @param data Pointer to parent.
+ *
+ * @return Neighbor data pointer if found, NULL if neighbor is not found.
+ */
+struct net_ipv6_nbr_data *net_rpl_get_ipv6_nbr_data(
```
Comment (tbursztyka) [inline subsys/net/ip/rpl.h]: I would break the line as:
struct net_ipv6_nbr_data *
net_rpl_get_ipv6_nbr_data(struct net_rpl_parent *parent);

**Review by tbursztyka — CHANGES_REQUESTED** [review]
small typo in 3rd commit : s/Sp/So I guess

**Review by jukkar — APPROVED** [review]
LGTM

**Review by tbursztyka — APPROVED** [review]

**Pull request merged**